Well, I'm still seeing the same exception as in https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/NPE.20when.20using.20quarkus.3Adev/near/268544477
```
Listening for transport dt_socket at address: 5005
Exception in thread "main" java.lang.RuntimeException: java.lang.NullPointerException
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:138)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: java.lang.NullPointerException
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:90)
        ... 1 more
```
I added some output right above the code you changed and I don't see it, so that part is not even invoked?

--- a/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java
@@ -955,7 +955,9 @@ public class DevMojo extends AbstractMojo {
                     .setRepositorySystem(repoSystem)
                     .setRemoteRepositories(repos)
                     .setRemoteRepositoryManager(remoteRepositoryManager)
-                    .setWorkspaceDiscovery(true);
+                    .setWorkspaceDiscovery(true)
+                    .setPreferPomsFromWorkspace(true)
+                    .setCurrentProject(project.getFile().toString());
 
             // if it already exists, it may be a reload triggered by a change in a POM
             // in which case we should not be using the original Maven session

--- a/devtools/maven/src/main/java/io/quarkus/maven/QuarkusBootstrapProvider.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/QuarkusBootstrapProvider.java
@@ -120,6 +120,7 @@ public class QuarkusBootstrapProvider implements Closeable {
             try {
                 return MavenArtifactResolver.builder()
                         .setWorkspaceDiscovery(mode == LaunchMode.DEVELOPMENT || mode == LaunchMode.TEST)
+                        .setCurrentProject(mojo.mavenProject().getFile().toString())
                         .setPreferPomsFromWorkspace(mode == LaunchMode.DEVELOPMENT || mode == LaunchMode.TEST)
                         .setRepositorySystem(repoSystem)
                         .setRepositorySystemSession(mojo.repositorySystemSession())

